lib/identity_cache/cached/association.rb

```diff
@@ -0,0 +1,43 @@
+module IdentityCache
+  module Cached
+    class Association # :nodoc:
+      def initialize(name, inverse_name:, reflection:)
+        @name = name
+        @reflection = reflection
+        # TODO: See if this can be replaced with `reflection.inverse_of.name`?
+        @inverse_name = inverse_name
+      end
+
+      attr_reader :name, :reflection
+      attr_accessor :inverse_name
```

Comment:
I don't like that this is an accessor, but we try to set it in the expiration hook code if it isn't already set. Seems like this could be made better.